pkg/actuators/machine/utils.go

```diff
@@ -83,6 +83,56 @@ func getExistingInstances(machine *machinev1.Machine, client awsclient.Client) (
 	})
 }
 
+func getExistingInstanceByID(id string, client awsclient.Client) (*ec2.Instance, error) {
+	return getInstanceByID(id, client, []*string{
+		aws.String(ec2.InstanceStateNameRunning),
+		aws.String(ec2.InstanceStateNamePending),
+		aws.String(ec2.InstanceStateNameStopped),
+		aws.String(ec2.InstanceStateNameStopping),
+		aws.String(ec2.InstanceStateNameShuttingDown),
+	})
+}
+
+// getInstanceByID returns the instance with the given ID if it exists.
+func getInstanceByID(id string, client awsclient.Client, instanceStateFilter []*string) (*ec2.Instance, error) {
```

Comment:
we could move out the hardcoded `clusterID` filter and reuse `getInstances` here

Reply:
That method searches by name tag though. We'd have to change it to just take a generic set of filters or something.